src/lein_monolith/task/fingerprint.clj

```diff
@@ -0,0 +1,352 @@
+(ns lein-monolith.task.fingerprint
+  (:require
+    [clojure.data]
+    [clojure.edn :as edn]
+    [clojure.java.io :as jio]
+    [clojure.set :as set]
+    [clojure.string :as str]
+    [leiningen.core.main :as lein]
+    [leiningen.core.project :as project]
+    [lein-monolith.dependency :as dep]
+    [lein-monolith.target :as target]
+    [lein-monolith.task.util :as u]
+    [multihash.core :as mhash]
+    [multihash.digest :as digest]
+    [puget.color.ansi :as ansi]
+    [puget.printer :as puget])
+  (:import
+    (java.io
+      File
+      PushbackInputStream)))
+
+
+;; ## Options
+
+(def selection-opts
+  (assoc target/selection-opts :upstream 0 :downstream 0))
+
+
+;; ## Hashing projects' inputs
+
+(def ^:private ->multihash
+  "Globally storing the algorithm we use to generate each multihash."
+  digest/sha1)
+
+
+(defn- aggregate-hashes
+  "Takes a collection of multihashes, and aggregates them together into a unified hash."
+  [mhashes]
+  ;; TODO: is there a better way to do this?
+  (if (= 1 (count mhashes))
+    (first mhashes)
+    (->> mhashes
+         (map mhash/base58)
+         (sort)
```

Comment:
This feels wrong to me - presumably these hashes correspond to projects or source files, and the order may be significant. For example, if `src/foo/a.clj` hashes to `ABC` and `src/foo/b.clj` hashes to `XYZ`, that is not the same as if `a.clj` hashes to `XYZ` and `b.clj` to `ABC`.

Reply:
For each file I am including the filename in the content that gets hashed (see the implementation of `hash-file`), so the case you're describing is accounted for. Therefore I think having an "unordered" aggregation here is appropriate, and it also makes sure that the order in which we happen to iterate over each file doesn't impact the hash.